github-action.py

```diff
@@ -740,6 +746,211 @@ def upload(self):
                 pass
 
 
+class AutoActionISO(BaseAutoAction):
+    def __init__(
+        self,
+        builder_dir,
+        config,
+        iso_timestamp,
+        state_dir,
+        commit_sha,
+        repository_publish,
+        local_log_file,
+    ):
+        super().__init__(
+            builder_dir=builder_dir,
+            config=config,
+            state_dir=state_dir,
+            commit_sha=commit_sha,
+            repository_publish=repository_publish,
+            local_log_file=local_log_file,
+        )
+
+        self.timeout = self.config.get("timeout", 21600)
+
+        config_iso = self.config.get("iso", {})
+        config_iso["version"] = commit_sha
+        self.config.set("iso", config_iso)
```

Comment:
There should be also some way to set `is-final` (for example when building RC vs final release). I'm okay-ish with doing that implicitly based on the version (whether it contains rc/alpha/beta) - for example configurable list of strings to look in version to force is-final=false.